### PR TITLE
feature: add ds18 sensor

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description

this is the implementation of  the ds18 temperature sensor

### Testing procedure

make -C test/driver_ds18 BOARD=your_board PORT=/dev/your_port flash term

### Issues/PRs references

 this PR fixes the issue #74

### CODE REVIEW

File: firmware/peripherals/ds18_sensor/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,2 @@
+USEMODULE_INCLUDES_ds18_sensor := $(LAST_MAKEFILEDIR)/include
+USEMODULE_INCLUDES += $(USEMODULE_INCLUDES_ds18_sensor)
```
Comment (luisan00) [inline firmware/peripherals/ds18_sensor/Makefile.include:2]: no EOF

File: firmware/peripherals/ds18_sensor/include/ds18_sensor.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * @return  0 satisfactory result
+ *         -1 failed result
+ */
+int get_temperature_value(int16_t* output);
+
+#ifdef __cplusplus
+}
+#endif
+#endif /* DS18_SENSOR_H */
+/** @} */
```
Comment (luisan00) [inline firmware/peripherals/ds18_sensor/include/ds18_sensor.h:55]: No newline at EOF

File: firmware/peripherals/ds18_sensor/include/ds18_sensor.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * @brief  this function initializes the temperature sensor
+ * @param  pin  pin number where the data will be taken
+ * @return  0 satisfactory result
+ *         -1 failed result
+ */
+int init_temperature_sensor(int pin);
+
+/**
+ * @brief  function to get the sensor data
+ * @param[out]  output this variable will return the temperature value
```
Comment (luisan00) [inline firmware/peripherals/ds18_sensor/include/ds18_sensor.h:45]: Is this compiling correctly?
```
 * @param[out]  output this variable will return the temperature value
```

**Review by luisan00 — CHANGES_REQUESTED** [review]
looks good , only some details

**Review by eduazocar — APPROVED** [review]

**Review by luisan00 — APPROVED** [review]

**Pull request merged**